checks/runchecks.go

```diff
@@ -184,3 +188,55 @@ func processTemplates(ctx *context.Context, r *pkg.CheckResult) *pkg.CheckResult
 
 	return r
 }
+
+func PersistCheckResults(ctx dutyContext.Context, canaryID string, canary v1.Canary, results []*pkg.CheckResult) {
+	// Get transformed checks before and after, and then delete the olds ones that are not in new set
+	existingTransformedChecks, _ := db.GetTransformedCheckIDs(ctx, canaryID)
+	var transformedChecksCreated []string
+	// Transformed checks have a delete strategy
+	// On deletion they can either be marked healthy, unhealthy or left as is
+	checkIDDeleteStrategyMap := make(map[string]string)
+
+	// TODO: Use ctx with object here
+	logPass := canary.IsTrace() || canary.IsDebug()
+	logFail := canary.IsTrace() || canary.IsDebug()
+
+	for _, result := range results {
+		if logPass && result.Pass || logFail && !result.Pass {
+			logger.Infof(result.String())
+		}
+
+		transformedChecksAdded := cache.PostgresCache.Add(pkg.FromV1(result.Canary, result.Check), pkg.FromResult(*result))
+		transformedChecksCreated = append(transformedChecksCreated, transformedChecksAdded...)
+		for _, checkID := range transformedChecksAdded {
+			checkIDDeleteStrategyMap[checkID] = result.Check.GetTransformDeleteStrategy()
+		}
+	}
+
+	checkDeleteStrategyGroup := make(map[string][]string)
```

Comment:
I don't think we can reuse the transformation deletion logic here, With a webhook the existingTransformed checks are not relevant - We probably need to add a deleted_at value in TransformedCheckResult and in the transformaton function set deleted_at e.g. when an alert status is resolved.